tests/src/core/testqgsmatrix4x4.cpp

```diff
@@ -0,0 +1,176 @@
+/***************************************************************************
+  qgsmatrix4x4.cpp
+  --------------------------------------
+  Date                 : July 2023
+  Copyright            : (C) 2023 by Martin Dobias
+  Email                : wonder dot sk at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgstest.h"
+#include "qgsmatrix4x4.h"
+
+/**
+ * \ingroup UnitTests
+ * This is a unit test for the QgsMatrix4x4 class
+ */
+class TestQgsMatrix4x4 : public QObject
+{
+    Q_OBJECT
+  public:
+    TestQgsMatrix4x4() = default;
+
+  private slots:
+    void initTestCase();// will be called before the first testfunction is executed.
+    void cleanupTestCase();// will be called after the last testfunction was executed.
+
+    void testData();
+    void testIdentity();
+    void testVectorMultiply();
+    void testMatrixMultiply();
+
+  private:
+};
+
+//runs before all tests
+void TestQgsMatrix4x4::initTestCase()
+{
+}
+
+//runs after all tests
+void TestQgsMatrix4x4::cleanupTestCase()
+{
+}
+
+void TestQgsMatrix4x4::testData()
+{
+  // Initialization - row-major order
+  QgsMatrix4x4 m1( 10, 20, 30, 40,
```

Comment:
```suggestion
  QgsMatrix4x4 m1( 10.0, 20.0, 30.0, 40.0,
```

or simply "x.", but avoid implicit conversions